src/amdinfer/workers/rocal.cpp

```diff
@@ -0,0 +1,439 @@
+              depth = node["crop_depth"].GetUint();
+              is_output = node.HasMember("is_output") ? node["is_output"].GetBool() : false;
+              x = node["crop_pos_x"].GetFloat();
+              y = node["crop_pos_y"].GetFloat();
+              z = node["crop_pos_z"].GetFloat();
+              outputTensor = rocalCropFixed(handle_, inputTensor, width, height, depth, is_output, x, y, z);
+              break;
+          case RocalOperation::ResizeCropMirrorFixed:
+              width = node["dest_width"].GetUint();
+              height = node["dest_height"].GetUint();
+              is_output = node.HasMember("is_output") ? node["is_output"].GetBool() : false;
+              y = node["crop_h"].GetUint();
+              x = node["crop_w"].GetUint();
+              mirror = node.HasMember("mirror") ? node["mirror"].GetBool() : false;
+              p_mirror = rocalCreateIntParameter(mirror);
+              outputTensor = rocalResizeCropMirrorFixed(handle_, inputTensor, width, height, is_output, y, x, p_mirror);
+              break;
+          default:
+              std::string errorMessage = "Unrecognized operation";
+              std::cerr << errorMessage << std::endl;
+              throw invalid_argument(errorMessage);
+      }
+      tensors.push_back(outputTensor);
+  }
+
+  if (rocalGetStatus(this->handle_) != ROCAL_OK) {
+      std::string errorMessage = "Error while adding the augmentation nodes\n" + std::string(rocalGetErrorMessage(this->handle_));
+      throw runtime_error(errorMessage);
+  }
+
+  // Calling the API to verify and build the augmentation graph
+  if (rocalVerify(this->handle_) != ROCAL_OK) {
+      std::string errorMessage = "Could not verify the rocAL pipeline\n" + std::string(rocalGetErrorMessage(this->handle_));
+      throw runtime_error(errorMessage);
+  }
+}
+
+void RocalWorker::doInit(ParameterMap* parameters) {
+  // default batch size; client may request a change. Arbitrarily set to 64
+  const int default_batch_size = 64;
+
+  this->batch_size_ = default_batch_size;
+#ifdef AMDINFER_ENABLE_LOGGING
+  const auto& logger = this->getLogger();
+#endif
+  // stringstream used for formatting logger messages
+  std::string msg;
+  std::stringstream smsg(msg);
+  AMDINFER_LOG_INFO(logger, " RocalWorker::doInit \n");
+
+  if (parameters->has("batch")) {
+    this->batch_size_ = parameters->get<int>("batch");
+    this->ROI_xywh_.resize(this->batch_size_);
+  }
+  if (parameters->has("model")) {
+    this->model_path_ = parameters->get<std::string>("model");
+  }
+  else {
+    AMDINFER_LOG_ERROR(
+      logger, "RocalWorker parameters required:  \"model\": \"<filepath>\"");
+    // Throwing an exception causes server to delete this worker instance.
+    // Client must try again.
+    throw invalid_argument(
+      "model file argument missing from model load request");
+  }
+  
+  this->handle_ = rocalCreate(batch_size_, RocalProcessMode::ROCAL_PROCESS_CPU, 0, 1);
```

Comment:
probably you might want to take processing mode as a parameter as well for CPU/GPU usecase